themes/bootstrap3/templates/Recommend/SideFacets/hierarchical-facet.phtml

```diff
@@ -32,7 +32,16 @@ JS;
   ?>
   <?=$this->inlineScript(\Laminas\View\Helper\HeadScript::SCRIPT, $script, 'SET'); ?>
 <?php endif; ?>
-<li id="facet_<?=$this->escapeHtml($this->title)?>" class="jstree-facet"
+<?php
+  $truncateSettings = $this->facets_before_more < 1 ? false : [
+    'rows' => $this->facets_before_more,
+    'btn-class' => 'facet',
+    'wrapper-class' => false,
+    'wrapper-tagname' => 'div',
+  ];
+?>
+<div<?php if ($truncateSettings): ?> class="truncate-hierarchy" data-truncate="<?=$this->escapeHtml(json_encode($truncateSettings))?>"<?php endif; ?>>
```

Comment:
Is it legal to wrap a `<div>` around an `<li>` or does the `<li>` need to be directly inside the containing list? If the latter, maybe we should be applying the truncation settings directly to the `<li>` instead of building a container -- and if so, we may need to adjust the truncation selector settings.

Reply:
Just to follow up, I tried eliminating the div with this patch:

```
-<div<?php if ($truncateSettings): ?> class="truncate-hierarchy" data-truncate="<?=$this->escapeHtml(json_encode($truncateSettings))?>"<?php endif; ?>>
-  <li id="facet_<?=$this->escapeHtml($this->title)?>" class="jstree-facet"
-    data-source="<?=$this->escapeHtml($this->searchClassId)?>"
-    data-facet="<?=$this->escapeHtmlAttr($this->title)?>"
-    data-path="<?=$this->currentPath()?>"
-    data-exclude="<?=$this->allowExclude?>"
-    data-operator="<?=$this->recommend->getFacetOperator($this->title)?>"
-    data-exclude-title="<?=$this->transEscAttr('exclude_facet')?>"
-    data-sort="<?=$this->sortOptions ?? ''?>"
-    data-query="<?=$this->escapeHtmlAttr($urlQuery)?>"
-    data-query-suppressed="<?=$querySuppressed ? '1' : '0' ?>"
-    data-extra-fields="<?=$this->escapeHtml(implode(',', $extraUrlFields))?>">
-  </li>
-</div>
+<li id="facet_<?=$this->escapeHtml($this->title)?>"
+  class="jstree-facet<?php if ($truncateSettings): ?> truncate-hierarchy<?php endif; ?>"
+  <?php if ($truncateSettings): ?>data-truncate="<?=$this->escapeHtml(json_encode($truncateSettings))?>"<?php endif; ?>
+  data-source="<?=$this->escapeHtml($this->searchClassId)?>"
+  data-facet="<?=$this->escapeHtmlAttr($this->title)?>"
+  data-path="<?=$this->currentPath()?>"
+  data-exclude="<?=$this->allowExclude?>"
+  data-operator="<?=$this->recommend->getFacetOperator($this->title)?>"
+  data-exclude-title="<?=$this->transEscAttr('exclude_facet')?>"
+  data-sort="<?=$this->sortOptions ?? ''?>"
+  data-query="<?=$this->escapeHtmlAttr($urlQuery)?>"
+  data-query-suppressed="<?=$querySuppressed ? '1' : '0' ?>"
+  data-extra-fields="<?=$this->escapeHtml(implode(',', $extraUrlFields))?>">
+</li>
```

...but it didn't work. I don't have time to troubleshoot further this week, but maybe it's not worth the effort. It appears that we already have facet entries as `<li>` tags without containing lists, and some of them are wrapped in `<div>` tags for collapsing, so if this is wrong, it's not more wrong than what we're already doing. This may need to be revisited, but that's probably a separate issue.

Reply:
In mozilla developer (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/li ) it is stated that permitted parents for `<li>` are `<ul>`, `<ol>`, or `<menu>` element. "Though not a conforming usage, the obsolete `<dir>` can also be a parent." So it seems like div is not allowed as parent. I assume this would need to change at several places.

Reply:
> Is it legal to wrap a `<div>` around an `<li>` or does the `<li>` need to be directly inside the containing list? If the latter, maybe we should be applying the truncation settings directly to the `<li>` instead of building a container -- and if so, we may need to adjust the truncation selector settings.

Hi there, just chanced upon this:

As @j4lib mentions, you mustn't wrap <li> in <div> -- however, we were asked by our accessibility testing firm, to **consistently style all facets using <ul> since they are lists**. This is a bigger chunk which needs changes in several files, so definitely something for VuFind 9, imo.

Reply:
Thanks, @ckaz. I think there's a very strong possibility that we can just change the `<li>` to a `<div>` in this case and the hierarchical facets will be okay (I can try this when I get back next week, if nobody objects or does it in the meantime) -- I'm pretty sure that the jstree library used for rendering them already uses lists internally and is reasonably accessible. Then, as you say, the other project is reformatting the other facets as lists, which is a separate project that would be appropriate for 9.0 -- I welcome a PR on that if anyone has time.

Reply:
> Then, as you say, the other project is reformatting the other facets as lists, which is a separate project that would be appropriate for 9.0 -- I welcome a PR on that if anyone has time.

@Demian: I think we have this implemented well enough to warant a PR on this soon!